server/src/middleware/auth.js

```diff
@@ -0,0 +1,20 @@
+const jwt = require("jsonwebtoken");
+const config = require("../config");
+
+module.exports = {
+    async loginRequired(req,res,next) {
+        try{
+            const token = req.headers.authorization.split(" ")[1];
+            jwt.verify(token, config.privateKey, (err, decoded) => {
+                const d = new Date();
+                if (decoded && Number(decoded.exp)*1000>d.getTime()) {
+                    next();
+                } else {
+                    return res.status(400).json({error:"Please Log In First"});
```

Comment:
This signifies that the token was correct but the time limit expired. Please correct me if wrong. 
It  would be better if the status is 401 rather than 400 to signify that auth is invalid.

Reply:
I'll update accordingly.